### PR TITLE
docs(relay-kit): Add Safe4337Pack docs

### PR DESCRIPTION
**Pull request by yagopv**
Added documentation for the new `Safe4337Pack`:

- Added guide 
- Added reference

### CODE REVIEW

File: pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+```typescript
+const PRIVATE_KEY = "0x...";
+
+const provider = new ethers.JsonRpcProvider("https://rpc.ankr.com/eth_sepolia");
+const signer = new ethers.Wallet(PRIVATE_KEY, provider);
+const ethersAdapter = new EthersAdapter({
+  ethers,
+  signerOrProvider: signer,
+});
```
Comment (germartinez) [inline pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx:59]:
```suggestion
const PRIVATE_KEY = '0x...'

const provider = new ethers.JsonRpcProvider('https://rpc.ankr.com/eth_sepolia')
const signer = new ethers.Wallet(PRIVATE_KEY, provider)
const ethersAdapter = new EthersAdapter({
  ethers,
  signerOrProvider: signer
})
```

File: pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+```typescript
+// Create some transactions
+const transaction1 = { to, data, value };
+const transaction2 = { to, data, value };
+
+const transactions = [transaction1, transaction2];
+
+// Create the transaction batch bundled in a SafeOperation object
+const safeOperation = await safe4337Pack.createTransaction({ transactions });
```
Comment (germartinez) [inline pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx:112]:
```suggestion
// Create some transactions
const transaction1 = { to, data, value }
const transaction2 = { to, data, value }

const transactions = [transaction1, transaction2]

// Create the transaction batch bundled in a SafeOperation object
const safeOperation = await safe4337Pack.createTransaction({ transactions })
```

File: pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+Calling the `createTransaction()` method returns a `SafeOperation` object, which contains transaction batch data encoded by the Safe multisend contract. If the Safe account doesn't exist, the `initCode` field of the user operation will hold the Safe account creation bytecode ready to be sent to the bundler. As the bundler processes the user operation, this code is used to deploy the Safe account before executing the transaction batch.
+
+### Sign the SafeOperation
+
+Before sending it to the bundler, it's essential to sign the `SafeOperation` with the signer associated to the `EthersAdapter` instance. This generates a signature that will authorize the execution of the transaction batch.
+
+```typescript
+const signedSafeOperation = await safe4337Pack.signSafeOperation(safeOperation);
```
Comment (germartinez) [inline pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx:122]:
```suggestion
const signedSafeOperation = await safe4337Pack.signSafeOperation(safeOperation)
```

File: pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+### Execute the transaction
+
+Finally, once the `SafeOperation` is signed, you can forward the underlying user operation to the bundler.
+
+```typescript
+const userOperationHash = await safe4337Pack.executeTransaction({
+  executable: signedSafeOperation,
+});
```
Comment (germartinez) [inline pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx:132]:
```suggestion
const userOperationHash = await safe4337Pack.executeTransaction({
  executable: signedSafeOperation
})
```

File: pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+```typescript
+let userOperationReceipt = null;
+
+while (!userOperationReceipt) {
+  await new Promise((resolve) => setTimeout(resolve, 2000)); // Wait 2 seconds before checking the status again
+  userOperationReceipt = await safe4337Pack.getUserOperationReceipt(
+    userOperationHash
+  );
+}
```
Comment (germartinez) [inline pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx:149]:
```suggestion
let userOperationReceipt = null

while (!userOperationReceipt) {
  await new Promise((resolve) => setTimeout(resolve, 2000)) // Wait 2 seconds before checking the status again
  userOperationReceipt = await safe4337Pack.getUserOperationReceipt(
    userOperationHash
  )
}
```

File: pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+In addition, you can use the `getUserOperationByHash()` method with the returned hash to retrieve the user operation object you sent to the bundler.
+
+```typescript
+const userOperationPayload = await safe4337Pack.getUserOperationByHash(
+  userOperationHash
+);
```
Comment (germartinez) [inline pages/sdk/relay-kit/guides/how-to-safe4337pack.mdx:157]:
```suggestion
const userOperationPayload = await safe4337Pack.getUserOperationByHash(
  userOperationHash
)
```

File: pages/sdk/relay-kit/reference/safe-4337-pack.md [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+```typescript
+const safe4337Pack = await Safe4337Pack.init({
+  ethersAdapter,
+  rpcUrl,
+  bundlerUrl,
+  safeModulesVersion,
+  customContracts,
+  options,
+  paymasterOptions,
+});
```
Comment (germartinez) [inline pages/sdk/relay-kit/reference/safe-4337-pack.md:26]:
```suggestion
const safe4337Pack = await Safe4337Pack.init({
  ethersAdapter,
  rpcUrl,
  bundlerUrl,
  safeModulesVersion,
  customContracts,
  options,
  paymasterOptions
})
```

**Review by germartinez — CHANGES_REQUESTED** [review]
So far, this is a review of the code snippets.
I'll continue reviewing this guide

File: pages/sdk/relay-kit/guides/4337-safe.mdx [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
+## Prerequisites
+
+- [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+- A [Pimlico account](https://dashboard.pimlico.io) and an API key.
+
+## Install dependencies
+
+```bash
+yarn add ethers @safe-global/protocol-kit @safe-global/relay-kit@alpha
```
Comment (dasanra) [inline pages/sdk/relay-kit/guides/4337-safe.mdx:19]: We should publish a final version when the docs are ready.

**Pull request merged**